source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRegistration.mm

```diff
@@ -280,4 +281,142 @@ - (NSString *)getFeatureVersion:(NSString *)featureName
     return _featureRegistration;
 }
 @end
-;
+
+
+@interface ACRActionTargetBuildersList : NSObject
+
+@property ACRTargetCapability capability;
+
+- (instancetype)init:(ACRTargetCapability)capability;
+
+@end
+
+@implementation ACRActionTargetBuildersList {
+    NSDictionary<NSNumber *, ACRTargetBuilder *> *_builders;
+    NSMutableDictionary<NSNumber *, ACRTargetBuilder *> *_overwrittenBuilders;
+}
+
+- (instancetype)init:(ACRTargetCapability)capability
+{
+    NSNumber *openUrl = [ACOBaseActionElement getKey:ACROpenUrl];
+    NSNumber *submit = [ACOBaseActionElement getKey:ACRSubmit];
+    NSNumber *showcard = [ACOBaseActionElement getKey:ACRShowCard];
+    NSNumber *toggle = [ACOBaseActionElement getKey:ACRToggleVisibility];
+    NSNumber *unknown = [ACOBaseActionElement getKey:ACRUnknownAction];
+
+    _overwrittenBuilders = [[NSMutableDictionary alloc] init];
+
+    // target capability lists supported events and corresponding target builders
+    switch (capability) {
+        case ACRAction:
+            _builders = @{
+                openUrl : [ACRAggregateTargetBuilder getInstance],
+                submit : [ACRAggregateTargetBuilder getInstance],
+                showcard : [ACRShowCardTargetBuilder getInstance],
+                toggle : [ACRToggleVisibilityTargetBuilder getInstance]
+            };
+            break;
+        case ACRSelectAction:
+            _builders = @{
+                openUrl : [ACRAggregateTargetBuilder getInstance],
+                submit : [ACRAggregateTargetBuilder getInstance],
+                toggle : [ACRToggleVisibilityTargetBuilder getInstance],
+                unknown : [ACRUnknownActionTargetBuilder getInstance]
```

Comment:
>unknown : [ACRUnknownActionTargetBuilder getInstance] [](start = 16, length = 53)

Does this represent custom actions? Why is it only here for SelectAction but not for the case above?

Reply:
@RebeccaAnne , let me test it out the scenario and get back

Reply:
@RebeccaAnne, I checked the sample, and the reason for not having unknown action handler registration is that it is not needed. If you have a custom action, you need a custom renderer, and you can register the action handler there. The select action, on the other hand, is different since they may want to have a custom action type, but if the users use a default renderer such as a TextBlock renderer,  the renderer has to register the custom action handler, so it's only there in the select action. The users can still register their action handler builders if they wish.